### DISCUSSION TITLE
Unbricking

### DISCUSSION THREAD

**Discussion by mm84** [ludwig-v/wireless-carplay-dongle-reverse-engineering#72] [General]
Hi,
I recently brick my device after downgrade to 2020 version because I've stuck on 2021.03.06 version. I didn't know that procedure isn't allowed. Any chances to bring my box to live? 

Best regards,
Marcin

**Comment by mhkim90** [comment]
If your device doesn't show any led signal (red or green), you may can debrick your device by dumping a flash.
Check #59 .
I succeeded without desoldering.
![사진 2021  5  7  20 55 14](https://user-images.githubusercontent.com/70556680/119085143-1b148380-ba3e-11eb-99a8-e86750cb7a26.jpg)

**Reply by mm84** [reply]
Thanks mhkim90

Ok, i will try to make bypass, because i dont have this clips what you have. What kind of programmer do you have? Could you write a list things and apps I need to unbrick? Or maybe fast instruction for me and others? Thank you in advance for your answer

Update:
Bought programmer with SOIC8.

**Reply by mhkim90** [reply]
I used CH341A mini programmer with AsProgrammer.

**Reply by mm84** [reply]
Thanks, i'll try tomorrow.

**Reply by jsm174** [reply]
@mm84, fwiw, I used the same exact programmer above and tried to make a dump and ended up bricking a perfectly good unit. I tried to make a dump of another one, and now `sync` will never write anything back to the file system.

Documented here: https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/issues/8#issuecomment-742146406

I bought some replacement MX25L12835F chips, programmed one, but have yet to attempted to desolder / solder.

According to this, https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/59#discussioncomment-652159 you need to short circuit the xtal of the cpu. If you figure out how to do this, please share!

**Reply by mm84** [reply]
@jsm174  thanks

@mhkim90 did you short circuit the xtal of the cpu.?

**Reply by mhkim90** [reply]
@mm84 Though I didn't short circuit, I succeed in debricking two different devices.
But desolder is safe choice. 
Note that the version of debricked devices was 2020.xx.xx.

**Reply by mm84** [reply]
Thanks! I've made it! :) Without desldering. :) Thanks @mhkim90 and @jsm174 !

**Comment by Radias123** [comment]
> Hi,
> I recently brick my device after downgrade to 2020 version because I've stuck on 2021.03.06 version. I didn't know that procedure isn't allowed. Any chances to bring my box to live?
> 
> Best regards,
> Marcin

Hi mate, im in the same situation as you and im kinda noob to this thing. My dongle bricked during an update and its dead, no light. I have a big favor to ask, can you please guide me how to do this thing? like a step by step guide if i get all the tools. I dont want to spend another 100$ on the dongle..
Thank you
Robert.

**Comment by zkSac** [comment]
> If your device doesn't show any led signal (red or green), you may can debrick your device by dumping a flash.
> 
> Check #59 .
> 
> I succeeded without desoldering.
> 
> ![사진 2021  5  7  20 55 14](https://user-images.githubusercontent.com/70556680/119085143-1b148380-ba3e-11eb-99a8-e86750cb7a26.jpg)
> 
> 

Hello, I also try to do the same, could you share the .bin file you used? Thanks

**Reply by mhkim90** [reply]
I used bronken's dump in [Flash_Dump](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump).

**Reply by leonardosalles** [reply]
hello guys, I will try the same steps, which software did you used to install the flash dump via programmer?

**Reply by mhkim90** [reply]
@leonardosalles I used 'asprogrammer_1.4.1'.

**Reply by giuseppesodano** [reply]
![61B7D16A-22B5-4098-9C5C-96ABD40F4434](https://user-images.githubusercontent.com/97917637/149835797-bc4e91bb-7dd7-4ce3-bd1b-aecd42296b24.jpeg)
Hi could someone help me to solve this when trying to flash ?  Thankkssss a lot 
![Uploading F80AF429-866D-4E88-8AC5-823520262EE2.jpeg…]()

**Reply by giuseppesodano** [reply]
Here is my wireless carlinkit 
![1F6C42EE-450C-4C65-BDDB-CD16D2767BB7](https://user-images.githubusercontent.com/97917637/149839324-531fd249-32a8-448b-88c4-e520fed8f266.jpeg)

**Comment by leonardosalles** [comment]
Is there anyone with the MX25L12835F_FW_20210331.bin dump? It seems that @brokens dump is related to Green UI version and my one is the RED UI version, so I got a "WIFI_BT" hardware error after flash @brokens dump :(

**Reply by mhkim90** [reply]
Someone in other community succeed flash the RED UI version using Green UI version dump, although I don't know the version used. So it is worth to try again.

**Reply by leonardosalles** [reply]
It is strange, my device is a little newer than the brokens dump, I think they changed the hardware so that is why I am receiving this error message regarding Wifi and BT

**Reply by Radias123** [reply]
im in the same situation as you, I have a big favor to ask, can you please guide me how to do this thing? like a step by step guide if i get all the tools. I dont want to spend another 100$ on the dongle..
Thank you

**Comment by mm84** [comment]
I've update from green to red and backward. Using only rollimage bin and changing names.

Best Regards,
Marcin

**Reply by leonardosalles** [reply]
Thanks for reply mate! So is just rename .img to .bin and flash on bios ? I am not sure which image I should use to flash, a little new to these hardware things.

**Comment by mm84** [comment]
No,
Leave alone img and bin 😊

Rename files for example - U2W_AUTOKIT_Update_2020.12.25.1732_ROLLBACK.img to U2W _Update_2020.12.25.1732.img.

Best Regards,
Marcin

**Comment by mm84** [comment]
I've flashed brand new box which come in small box - old comes in large.

Best Regards,
Marcin

**Comment by mm84** [comment]
I've using program CH341a Programmer for Windows and dongle [Image.jpeg]

I don't remember exactly. I try once and I did it.

Best Regards,
Marcin
________________________________

**Reply by Radias123** [reply]
Hi, i flashed today the dump file, now the adapter is alive!!! when i put it in a wall brick, i get green light. What do i do now?

**Comment by mm84** [comment]
Prepare usb drive and do everything wiht Ludwig instruction.

Best Regards,
Marcin

**Reply by Radias123** [reply]
Thx. Which file should i flash? can you point it? i think the adapter is v2 with green.

**Comment by mm84** [comment]
U2W_Update_2020.11.10.2138_ROLLBACK.img


Best Regards,
Marcin

**Reply by Radias123** [reply]
New problem..when i inserted the adapter into the power brick after flash, the light was green, now when i wanted to update and inserted into the power brick again, i have no more light. Any ideas?

**Comment by Radias123** [comment]
i flashed the dump file but no red light when i put the adapter in the brick, just solid green light. Anyone has any ideas?

**Comment by bsenseman** [comment]
Ok, I successfully flashed the chip with 20200804 firmware on my CPlay2Air adapter dated 2019-12-25.  No lights displayed during flashing; when I plugged into a wall-wort; I got slow red/blue alternating flash that speed up and eventually created only a dim blue.  No wifi, tried in car, nothing.  Reconnected to CH341A and the blue LED came on; reflashed.  Anybody know what the slow to fast is?  Recovery mode?  Should I attach a USB with firmware?  I know the blue/red did flash when I updated firmware previously.  Would really like to unbrick this dongle.

**Comment by ludwig-v** [comment]
Flash @bronken dump (MX25L12835F_FW_20210306.bin), it worked on 3 bricked CPlay2air units I had but now they are all blocked on activation so it is alive but useless 

The device will flash red & blue/green after powering it (the device is updating the iMX DTB) during about 5/10 seconds and you will see it alive within 15 seconds ;)

### Diagnostic

Fixed blue/green LED = Blocked on uBoot : the flash is possibly dead
Blue/green LED (few ms only, hard to see) at powering then no LED at all = Blocked on boot / possibly failing flash

**Reply by bsenseman** [reply]
Wow, 2 for 2 today!  Success!  The Activation screen through me for a loop initially, and then the car wifi interfered, but in the end; we have wireless carplay again!  Thank You.

**Reply by bsenseman** [reply]
If you are stuck on activation, all I did was scan the displayed QR code (takes you to 192.168.50.2) and once the web interface showed on my phone I got an "Activation Success!!!!" message on the radio display,  I did have a solid red LED during this; then it went to flashing blue once I had CarPlay.

**Reply by Didi25256** [reply]
Flash this file to get through the activation @ludwig-v 
U2W_Update_2020.11.10.2138_ROLLBACK.img

**Comment by dmavromatis** [comment]
Hi - I ordered a CH341A kit from Amazon... I want to install the latest CFW... 

- [x] Using AsProgrammer to dump and flash the firmware.
- [ ] @ludwig-v Which CFW version should we use to flash directly onto chip (this will be for V3.0 device)?

**Comment by dmavromatis** [comment]
Did a few successful direct flash of chip on a v2 dongle.   Then I tried to dump V2 CFW 2021.08.24 to BIN and flash V3 unit but it didn't work... I get a "Hardware Error: Wi-Fi BT". It's running the CFW, just giving me the Wi-Fi error... Let me know if that should work or if you need to dump a special version that works on V3 devices.   Looks like the bcm drivers are not being used since I have a dump from the v2 dongle.

**Reply by dmavromatis** [reply]
Oh, the good news, the CFW is running on the V3 -- so I can use the U2W.sh script, etc... so I'm wondering if there is a way to force it to use the BCM drivers (unless my dump is corrupt and the hardware error is a red herring).

**Comment by zkSac** [comment]
Carlinkit 2.0 CPC200-U2W From Amazon for AUDI, Volkswagen, Honda Civic/Mazda CX5, Chevrolet, Toyota, KIA, ETC 
.bin extracted. Upload to chip with EEPROM reader to unbrick your dongle
https://www.mediafire.com/file/ypy65oonce80ix4/CPC200-U2W.bin
Need more info? let me know. @ludwig-v pls can u upload to dump_flash
![IMG_0139](https://user-images.githubusercontent.com/27929631/142274285-471ec3ec-6bd9-4b9e-852f-6703d224ff72.png)
![IMG_0140](https://user-images.githubusercontent.com/27929631/142274288-97192c73-5944-48c1-b2ef-44ea1df59fa3.png)
![IMG_0141](https://user-images.githubusercontent.com/27929631/142274289-54a64579-7106-4296-8329-e22b2b71f4da.png)
![IMG_0145](https://user-images.githubusercontent.com/27929631/142274290-2cde3da4-ae97-4458-868e-38284cd06295.png)
![IMG_0148](https://user-images.githubusercontent.com/27929631/142274293-6e7632d8-2405-4c5f-863f-bfa95f91eeb6.png)
![IMG_0149](https://user-images.githubusercontent.com/27929631/142274294-1f454e47-c3a4-42a2-8683-ba7aafd5f8c9.png)

**Comment by italoc** [comment]
I have a car2play device death after a firmware upgrade, now no led on device, if I have understood I need a programmer and flash dump, after that I can ref flash firmware with a usb pen drive connected to car2play, is correct?

**Reply by goldelehr** [reply]
Hi @italoc ,

i was in the same situation as you. With the above mentioned CH341A board + ASProgrammer i was able to flash @brokens dump. After that my device was working again. I did not flash anything afterwards with an USB pen drive.

**Reply by italoc** [reply]
> Hi @italoc ,
> 
> i was in the same situation as you. With the above mentioned CH341A board + ASProgrammer i was able to flash @brokens dump. After that my device was working again. I did not flash anything afterwards with an USB pen drive.

i have solved with  CH341A board + ASProgrammer, flash bios with @brokens dump and, i have upgraded the firmware with USB pen drive to the latest version

**Comment by jailbird777** [comment]
So I unbricked my older 2.0 board using the dumped firmware. The downside is when using the posted dumped firmware, it doesn't seem to use the AUTOKIT stuff anymore, is this a known issue?

**Reply by ludwig-v** [reply]
You just have to change filenames to get back to AUTOKIT version (U2W_AUTOKIT_Update.img to U2W_Update.img) so it is not an issue at all.

By the way: There is also a flash dump with latest Custom Firmware (which is using the [theme of th1efx](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Custom_Firmware/Themes/AppleLike)) in the [Flash_Dump](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump) directory

**Comment by michas121** [comment]
![IMG_2923](https://user-images.githubusercontent.com/1313784/150129401-55274dd1-1054-4802-b911-d4ba452e258b.jpg)
![IMG_2924](https://user-images.githubusercontent.com/1313784/150129405-f3c33a13-10a4-482d-a324-f9a9bf19c3a7.jpg)
![IMG_2925](https://user-images.githubusercontent.com/1313784/150129407-2216bcc7-8fab-439f-aa75-9bdc6a630c07.PNG)
![IMG_2926](https://user-images.githubusercontent.com/1313784/150129410-7219667a-2200-43ac-b536-139ec14aff02.PNG)

My Carlinkit from Aliexpress starts to freeze. I tried to update via Autokit app and it brick. I unsolder a BIOS and flash ,,MX25L12835F_FW_20210306.bin" from bronken. After plug to OTG it starts to flash blue/red quickly and then turn right. As you can see in the photo I can connect to wifi and change settings. As you can see my android tablet identify Carlinkit as Apple iPhone and Autokit app doesnt see this device. Ive try making USB Stick with ,,U2W_Update_2020.11.10.2138_ROLLBACK.img" with U2W_Update, U2W_Autokit_Update, Auto_Box_Update but no flash - all time red light solid. Ive tried also Rollback via WiFi 192.168.50.2 to this version - with succes but still same issue - it doesnt detected by Autokit and its useless. What can I do? Ive tried another dumbs but with another dumbs LED dont flash or was blue solid :(

**Reply by ludwig-v** [reply]
You flashed a device made for Android with Carplay made version, you must [flash the correct image](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/59#discussioncomment-1679085)

**Reply by michas121** [reply]
Shiet, i've broke one pin from chip. It is possible to buy a new ,,MX25L12835F" then flash with this firmware and solder? Can I use some simillar chip (which one?) becouse in my country its hard to get ,,MX25L12835F"

**Reply by michas121** [reply]
UPDATE: I've bought new ,,MX25L12835F", flashed with ROM from ludwig-v and its working! Now I have to test it in car if it not crash, freeze etc. Do you have maybe what software and what device version is best?

**Reply by Vitalic66** [reply]
Do you remember which firmware you used?

I guess I got the same fake dongle.

Before I bricked it, it had the version 2020.06.04.0218 installed. I also made a dump before I flashed another firmware but didnt desolder IC.

Reflashing the IC doesnt work. LED stay off.

Also the mentioned firmware from „flash the correct image“ does not work.

**Comment by lukup77** [comment]
I've bricked ma v3 dongle - red UI - and now I'm trying to upload dump v3 firmware with ch341a.
But after it flash few times witch green and red led and it's brick again.
Have you got a working dump form v3 unit ?

**Reply by ZqinKing** [reply]
> I've bricked ma v3 dongle - red UI - and now I'm trying to upload dump v3 firmware with ch341a. But after it flash few times witch green and red led and it's brick again. Have you got a working dump form v3 unit ?

me too...

**Reply by lukup77** [reply]
I've managed how to fix it. After flash CFW dump for v3 device I upgraded it with U2W_AUTOKIT_Update_2021.08.24.1817.img file by usb dongle.
After update it works.

**Comment by giuseppesodano** [comment]
hi, i tried right now with the right customfirmware  : U2W_Update_2020.11.10.2138_ROLLBACK.img and renamed it into : Auto_Box_Update.img BUT wihtout success.  see pictures (in red) it seems that I haven't the right settings on my carlinkit wireless. could you please help me ? I'm getting crazy :)

regards
![IMG_3240](https://user-images.githubusercontent.com/97917637/151708087-2008d3f6-d9e0-466f-a498-c21670647547.jpg)
![IMG_3241](https://user-images.githubusercontent.com/97917637/151708091-a63fffcc-b965-4c50-bec5-ae2d70599943.jpg)
,

**Comment by michas121** [comment]
Hello! I've flashed ,,MX25L12835F_FW_20210306_retail.BIN" and its works very well about month. Now it doesn't work and when I go to 192.168.50.2 I see screen ,,Product Activation". Batch: 20220215. I need to put Account and Password? What the heck? Ive don't make any update manually.

**Comment by jamesgreystone** [comment]
Any chance someone can do a step by step write up of how do do this procedure. Would be great to revive my 150.00 Brick.

Much Appreciated.

**Reply by itsandr00** [reply]
First and foremost - PATIENCE. If you are not patient and do not take your time following the steps, you will mess this process up!

These instructions work for CPC200-U2W Plus V2, non-custom firmware
1. Buy CH341A mini programmer exactly as shown photo posted by mhkim90 above
2. DL @bronken dump file from: https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump
3. DL AsProgrammer_1.4.1; https://github.com/nofeletru/UsbAsp-flash/releases - known version to work!
4. Install CH341Serial-Driver; CH341SER.EXE - Note: USB A uses the SERIAL protocol
5. Assemble and connect the programmer exactly as shown in mhkim90's photo above. Look at it carefully and make sure all pieces are put together and the red wire is connected _**exactly**_ the same as in the photo. Be sure to plug into a USB 3.0 port in order to generate enough power. Once plugged into USB port and everything is correctly in place, the red power led on the CH341A USB board should light up. If you have no light, it means you have a short, disconnect and try again.
6. If all is good, open AsProgrammer and click Hardware at the top and make sure CH341A is selected. Next, press the button with the green question mark. A list of Macronix chip IDs should show up, double click MX25L12835F and then x to close the box.
7. Click the folder button to find and open the @bronken flash dump file you downloaded from 
https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump
8. Next, follow this sequence EXACTLY: Unprotect -> Erase -> Program -> Verify
    Unprotect: click the button with the padlock (NOT the black arrow)
    Erase: click the big red X to erase existing code from chip memory
    Program: click the red arrow pointing left (NOT the black arrow)
    Verify: Wait for programming to finish -- this step will also verify the file.

Congratulations - you are done! Now follow the steps to upload the rollback firmware
https://www.carlinkitfactory.com/u2w-plus-upgrade-firmware-offline-for-carlinkit-cpc200-u2w-u2w-plus-dongles-firmware-upgrade-by-usb-flash-drive/

DL U2W_Update_2020.11.10.2138_ROLLBACK.img and upload to a FAT32 USB stick.
https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Firmware

**Reply by Nica-Alex** [reply]
On step 5, the red power led should light up just on the programmer or on the Carlinkit dongle also?
I can get the dongle red led light up only when I touch the SOIC clip pins 1-4, when I touch the other 5-8 pins the led turns off.
I get ```ID(9F): FFFFFF(Unknown)
ID(90): FFFF(Unknown)
ID(AB): FF(Unknown)
ID(15): FFFF(Unknown)``` when pressing the green question mark button.
Any thoughts?

**Reply by Crewski** [reply]
It took me several tries to get the clip aligned on the chip properly. I’d try it a few more times to see if you can get it to read it. I had never used the clip before.

**Reply by Nica-Alex** [reply]
I've tried it a lot of times, everytime with the same result.
Btw, I had to install the Parallel driver, not the Serial one. Windows doesn't recognize the Serial driver when trying to identify my unknown device.

**Reply by SvenK84** [reply]
I had exactly the same issue until 2 hours ago when I finally got it working. For me the solution was to check the connection from each SOIC pin to the appropriate contact of the clip's plug via my multimeter. I used thin probes like these: https://www.amazon.de/gp/product/B087TNRM5L/ 

As soon as I successfully verified all connections I got AsProgrammer running and have a functioning Carlinkit again.
Thank you for everybody's work.

**Reply by Nica-Alex** [reply]
Verified the pins with the multimeter and everything is fine.
So, in order to be discoverable in AsProgrammer, the red power led from the Carlinkit dongle should light up and stay lit, if the connection to the chip is correct, right?

**Reply by itsandr00** [reply]
The red light on the CH341A USB board should light up. I noticed the red light on my dongle lighting up, but that was only after I had successfully flashed my dongle, removed it, uploaded the rollback firmware via USB and then plugged it back in to my computer, etc, etc.

**Comment by jailbird777** [comment]
Also, if you have a Rasperry Pi (or similar SBC), some jumper wires, and a SOIC clip, you can use its SPI bus and flashrom to ubrick the unit also. It's not quite as plug-and-play as the CH341A, but it does work :).

**Reply by pavelrevak** [reply]
> Also, if you have a Rasperry Pi (or similar SBC), some jumper wires, and a SOIC clip, you can use its SPI bus and flashrom to ubrick the unit also. It's not quite as plug-and-play as the CH341A, but it does work :).

Here is small tutorial for flashing using raspberry pi: https://github.com/pavelrevak/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump

I successfully unbricked and updated my device

_pull-request is created_

**Comment by mm84** [comment]
Replace chip.

Best Regards,
Marcin

**Comment by Converseallstar95** [comment]
I think I have the v3 (idk how to check), and it bricked during a software update. I found this thread when trying to figure out how to fix it. I bought the USB serial adapter and reprogrammed the bios according to the instructions here (https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/72#discussioncomment-2315150), but used the .bin file mentioned here (https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/130#discussioncomment-2324344). Everything seemed to go ok, but now when I startup my car, it loads CarPlay and shows "waiting activation" and a SSID/password; however, that wifi network never shoes on my phone to allow me to connect. Any ideas?

**Reply by lukup77** [reply]
> I think I have the v3 (idk how to check), and it bricked during a software update. I found this thread when trying to figure out how to fix it. I bought the USB serial adapter and reprogrammed the bios according to the instructions here (https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/72#discussioncomment-2315150), but used the .bin file mentioned here (https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/130#discussioncomment-2324344). Everything seemed to go ok, but now when I startup my car, it loads CarPlay and shows "waiting activation" and a SSID/password; however, that wifi network never shoes on my phone to allow me to connect. Any ideas?

I've managed to fix it. After flash CFW dump for v3 device I upgraded it with U2W_AUTOKIT_Update_2021.08.24.1817.img file by usb dongle.
After update it works without activation issue.

**Reply by Converseallstar95** [reply]
> > I think I have the v3 (idk how to check), and it bricked during a software update. I found this thread when trying to figure out how to fix it. I bought the USB serial adapter and reprogrammed the bios according to the instructions here ([#72 (reply in thread)](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/72#discussioncomment-2315150)), but used the .bin file mentioned here ([#130 (comment)](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/discussions/130#discussioncomment-2324344)). Everything seemed to go ok, but now when I startup my car, it loads CarPlay and shows "waiting activation" and a SSID/password; however, that wifi network never shoes on my phone to allow me to connect. Any ideas?
> 
> I've managed to fix it. After flash CFW dump for v3 device I upgraded it with U2W_AUTOKIT_Update_2021.08.24.1817.img file by usb dongle. After update it works without activation issue.

Thanks so much! I used 2021.08.24.1817 and named it "U2W_AUTOKIT_Update.img" and the update took. Went to my car and connected to wifi, then reset all settings from the web interface and it seems to be working once again!

**Comment by Waleed244** [comment]
I am having a constant red LED on Carlinkit 4.0 after flashing  its MX25L12835F with the ".bin" file,  as I had dead device before flashing(No LED at all)
I did format a USB stick  to FAT32 and named its firmware to U2AW_Update.img inserting it to the USB while being in power brick with no success ,getting "hardware error wifi bt" when connecting to the head unit ,any help ?

**Reply by Waleed244** [reply]
> I am having a constant red LED on Carlinkit 4.0 after flashing its MX25L12835F with the ".bin" file, as I had dead device before flashing(No LED at all) I did format a USB stick to FAT32 and named its firmware to U2AW_Update.img inserting it to the USB while being in power brick with no success ,getting "hardware error wifi bt" when connecting to the head unit ,any help ?

I have successfully unbricked my cpc200-cp2a using the BIOS clip and a raspberry pi , Now I am stuck with "need to complete activation "after flashing the FW with every boot and no option to activate ,neither the screen nor the web page ,any idea ?
![WhatsApp Image 2023-03-06 at 1 05 21 AM](https://user-images.githubusercontent.com/38866956/222991386-8fae2b31-cfe9-4553-b1dd-7eb4487ae607.jpeg)

**Comment by Theandhav** [comment]
Hello, bought a dongle used and it did not work i my car, but my parents car worked.
I never got 192.168.50.2 to work. Tried updating firmware with usb stick. It never worked, went straight to red light after short blue blink.
Suddenly it started to blink blue and red when i tried with 2020.08.04.1602_ROLLBACK. It was over to steady red light after only few seconds. 
Now it is only giving constant blue light immediately when powering. 
I want to try with CH341A, but don't even know which chip to connect. Seems like an old version. 
Does anyone have advice?
 
![IMG_0225](https://user-images.githubusercontent.com/97404290/219975146-9e1f9b5b-a0eb-416d-aaf8-201e0f8d255b.JPG)

**Reply by jklettb** [reply]
> Hello, bought a dongle used and it did not work i my car, but my parents car worked. I never got 192.168.50.2 to work. Tried updating firmware with usb stick. It never worked, went straight to red light after short blue blink. Suddenly it started to blink blue and red when i tried with 2020.08.04.1602_ROLLBACK. It was over to steady red light after only few seconds. Now it is only giving constant blue light immediately when powering. I want to try with CH341A, but don't even know which chip to connect. Seems like an old version. Does anyone have advice?
> 
> ![IMG_0225](https://user-images.githubusercontent.com/97404290/219975146-9e1f9b5b-a0eb-416d-aaf8-201e0f8d255b.JPG)

Hi the chip is the one marked as 25L12835F.

Best luck with it

**Reply by Theandhav** [reply]
Thanks!
I have connected the CH341a correctly I think, and configured AsProgrammer, but after trying a lot of different usb ports, extensions, connecting the clip, I'm only able to recognize the chip every 1/10 times i press the Read ID button. 
Most attempts to read anything is empty, but somtimes there is some data. 
![IMG_0275_(1)](https://user-images.githubusercontent.com/97404290/224422073-41f79eb4-8bbb-4764-adf7-c2ce1371b18c.jpg)
![attempts](https://user-images.githubusercontent.com/97404290/224422150-2dc95469-0fa4-4ff1-afd9-d6b9c8981df2.png)

**Comment by fajitamunch** [comment]
After many hours, I finally managed to flash the bin to my carlinkit mini device. I bought a CH341 kit and it took a lot of tinkering with the position of the clip to get it to work. I flashed the V3 bin, but I still have no LED light. It's weird because if I connect the CH341 clip in a normal manner, the green LED light comes on. I need to position it in a way for the light to turn off and then it will be detected and flash. 

If I then try to plug it into any socket using any wire, I have no LED light... Is the power delivery on my device dead? Or is it possible that because I had to have a weird clip position and the LED was off, that the bin somehow hasn't flashed properly (even though it verified ok)? Any help would be much appreciated!

Edit: After some research I think that the mini uses a different flash to the 3.0, even though it's called 3.0 mini... I can't see any other reason as to why this isn't working. I've tested with a multimeter and there's full power. If I short the LED pins, they turn on.

**Comment by ninenart** [comment]
I had a bricked Carlinkit A2A that I tried to unbrick by flashing with many Flash_Dump.bin files until I got the MX25L12835F_FW_20221226_V4.BIN and the Adapter worked again, but it converted to Apple Carplay and stuck on the Activation screen. Only CP2A V4 FW (U2AW_AUTOKIT_Update.img) can be used for update. 
I'd ask Carlinkit for the A2A FW, and they'd send me the (O2W_AUTOKIT_Update.img). Unfortunately, I cannot rename this file to convert it back. 

Any one can help to share the A2A flash dump ? or any suggestions?

**Comment by fajitamunch** [comment]
Does anyone have the carplay mini 2.0 flash file by any chance? I tried buying 2 from amazon to copy the flash files, but they were newer versions and were mini 3.0. The new ones have completely different PCBs and model names. I'm looking for the 'CPC200-Mini' model flash, not the 'CPC200-U2W' model.

If anyone has the flash file, I would really appreciate it!

**Reply by fajitamunch** [reply]
Following on from this, I bought 3 separate mini devices off amazon which advertised themselves as cpc200-mini and they were all CPC200-U2W. I even bought one off aliexpress that did the same thing to me. PLEASE if someone has a cpc200-mini, can you share the bin with me? I'm just trying to revive my dead device so it's not ا-waste. Thank you!

**Comment by professorflux** [comment]
Does anyone have for CPC200-2air bin version? I have bought CarlinkIt 5.0 CPC200-2air version and the firmware is corrupt no light on

**Reply by Levonson01** [reply]
Did you manage to fix this?

**Reply by vamzii** [reply]
any update on this please?

**Comment by waailasif3** [comment]
I just got my A2A dongle but it is not broadcasting any WiFi signal, BT works fine and connects too. What could be the issue? I tried flashing the O2W firmware sent by carlinkit support but it did not help ( I am not even sure if it flashed correctly or not).

**Comment by Quikeramos1** [comment]
I have managed to revive my Carlinkit V3 with the BIOS flashing. I leave you a tutorial of how I did my process. https://github.com/Quikeramos1/Unbrik-Carlinkit-V3

**Reply by dough29** [reply]
Nice work !
Do you know any available BIOS for CarlinKit 5.0 CPC200-2air / U2AC ?
I think the OTA update failed and now it's like dead.

**Reply by Levonson01** [reply]
Does this work with Carlinkit 5.0 also? Or do I need another bios/firmware??

**Comment by Quikeramos1** [comment]
Thank you so much! 

Currently I don't know of any place where I can get the BIOS to flash the V5. Have you tried to perform a firmware update via USB?

https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Firmware/U2AC￼

**Comment by Quikeramos1** [comment]
Each version of carlinkit has its own firmware and BIOS version. I know you can make the V3 work with the V4 through a CF.

**Comment by nikolasr200** [comment]
Hi,
i have a CarLinKit 4.0 which got bricked. I purchased a CH341 programmer (black) and try to flash firmware to cpu but no matter which programming software i used (AsProgrammer, NeoProgrammer, CH341 programmer) all fail to flash firmware, or freeze.
I read about a Voltage Hack for 3.3V chips on CH341 programmer, but according to the discussion over here, there shouldn't be a need for such hack in order to flash the firmware.
Am I doing something else wrong? or is the voltage hack necessary after all?
I would really appreciate your help!

**Comment by lawand44** [comment]
does anyone have flash dump for carlink v5 U2AC. I have tried the one in the U2AC flash dump folder for v5 but didn't work.

**Comment by ninja-** [comment]
newest bootloader(from mid 2024?), at least on U2AC, makes flash dumps non-transferable and flashing somebody else's flash dump will just brick your device. try flashing mid 2022 and mid 2023 dumps if you have any

@lawand44

**Reply by lawand44** [reply]
I don't know where to find mid 2022 and mid 2023 flash dumps which why I asked if anyone has a working flash dump for U2AC hardware

**Reply by ninja-** [reply]
try https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump/U2W 2022 files

**Comment by Zubastic** [comment]
Well, now I have u2ac sample with problem: it cutted off from power when upgrading. Userdata I downloaded from flash, but I guess that device was disabled when `AutoBox.img` was burning. So I tryed to burn my own u2ac dump. but nothing happends. Also u2w old dump, also nothing. Any ideas?

**Reply by gator8125** [reply]
What model do you have? It can autoload the oem update file thru usb (wait 3-5 minutes)

**Reply by Zubastic** [reply]
> What model do you have? It can autoload the oem update file thru usb (wait 3-5 minutes)

u2ac, how to do that?

**Comment by csturgill** [comment]
Not sure if this will help but i received this from carlinkit support. 

Hi there,

Please kindly download the firmware below to your USB drive, then try the method:

Link:  https://drive.google.com/file/d/1Ee6JdZXrmExV5tR7gIQ-CNP2Jm4X4Knv/view?usp=sharing 

1, prepare a 32G or less U disk, and right-click on the U disk, formatted as FAT32 format (4/8/16/32GB)

2, the img format firmware into the U disk (firmware name can not be modified, keep the name in the zip package)

3, find a charger plug to power up the product, and then insert the U disk into the box USB port to upgrade (do not upgrade in the car)

4, upgrade with red and green lights flashing, do not power off, wait until the bright red light or blue light is on, and then wait 3 minutes to unplug.


If the USB flash drive meets the size, and format, but the LED does not appear flashing phenomenon, please replace the USB flash drive.

This is the video tutorial: https://carlinkitcarplay.com/blogs/wireless-dongle-adapter/u2w-u2w-plus-firmware-upgrade-by-usb-flash-drive

Kind Regards
Nikki
